### PR TITLE
Add roadmap

### PR DESCRIPTION
**Pull request by albertpod**
This PR adds a roadmap section to the repository. 
There's a high-level roadmap in the form of the table, and there's additional link to the progress tracker of the issues which became public.
I've added multiple people to the reviewers. 
We should move as quickly as possible with this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +162,20 @@ Additionally, checkout our [video from JuliaCon 2023](https://www.youtube.com/wa
...
+Certainly! Here's an updated version of the roadmap section with added emojis:
+
+# Roadmap
+
+Our high-level project roadmap outlines the key milestones and focus areas over the next few quarters:
+
+| Q1/Q2 2024          | Q3/Q4 2024                | 2025                | 2026                                      |
+|---------------------|---------------------------|--------------------|-------------------------------------------|
+| 🧩 **Nested models with [GraphPPL.jl](https://github.com/biaslab/GraphPPL.jl)**     | 🛡️ **Robustness (NaN, Inf free)** | 🌐 **Stochastic Processes** | 🔄 **Automated inference with non-exponential family** 
+| 🔄 **Automated inference with [ExponentialFamily.jl](https://github.com/biaslab/ExponentialFamily.jl)**                 | 🧠 **Memory-efficiency** | 🚀 **Efficient message passing on a huge graph** | 📊 **Inference over graph structure**
```
Comment (bartvanerp) [inline README.md:174]: Regarding _Efficient message passing on a huge graph_, isn't this already one of the key selling points of RxInfer? Do you rather mean something like "Resource-adaptive inference"?

File: README.md [hunk 2]
```diff
@@ -161,6 +162,20 @@ Additionally, checkout our [video from JuliaCon 2023](https://www.youtube.com/wa
...
+
+For a more granular view of our progress and ongoing tasks, refer to our [project board](https://github.com/orgs/biaslab/projects/10/views/5).
```
Comment (bartvanerp) [inline README.md:176]:
```suggestion
For a more granular view of our progress and ongoing tasks, check out our [project board](https://github.com/orgs/biaslab/projects/10/views/5).
```

**Review by bartvanerp — CHANGES_REQUESTED** [review]
See feedback for my requested changes. The rest looks good!

File: README.md [hunk 3]
```diff
@@ -161,6 +162,20 @@ Additionally, checkout our [video from JuliaCon 2023](https://www.youtube.com/wa
...
 
+Certainly! Here's an updated version of the roadmap section with added emojis:
+
+# Roadmap
+
+Our high-level project roadmap outlines the key milestones and focus areas over the next few quarters:
+
+| Q1/Q2 2024          | Q3/Q4 2024                | 2025                | 2026                                      |
+|---------------------|---------------------------|--------------------|-------------------------------------------|
+| 🧩 **Nested models with [GraphPPL.jl](https://github.com/biaslab/GraphPPL.jl)**     | 🛡️ **Robustness (NaN, Inf free)** | 🌐 **Stochastic Processes** | 🔄 **Automated inference with non-exponential family**
```
[the remark this answers is not in the document]
Comment (albertpod) [reply README.md:173]: Are we planning to work on adding Dirichlet processes along with Gaussian Processes @ismailsenoz? I believe GPs will be available in Q4 of 2024 (@HoangMHNguyen ?)

**Review by wouterwln — APPROVED** [review]

**Review by bertdv — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (albertpod) [reply README.md:174]: Big graphs, not huge ones :) 
Anyway, it was supposed to be a triggering mechanism, but it's pretty slang. Resource-adaptive inference may sound better.

**Review by bartvanerp — APPROVED** [review]

**Pull request merged**